x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_view.tsx

```diff
@@ -0,0 +1,147 @@
+    tabId?: string;
+  }>();
+  useEffect(() => {
+    fetchEngine({ engineName });
+  }, [engineName]);
+
+  if (fetchEngineApiStatus.status === Status.ERROR) {
+    return (
+      <EnterpriseSearchEnginesPageTemplate
+        isEmptyState
+        pageChrome={[engineName]}
+        pageViewTelemetry={tabId}
+        pageHeader={{
+          pageTitle: engineName,
+          rightSideItems: [],
+        }}
+        engineName={engineName}
+        emptyState={<EngineError apiStatus={fetchEngineApiStatus} />}
+      />
+    );
+  }
+
+  const tabs: EuiTabbedContentTab[] = [
+    {
+      content: <div />,
+      id: EngineViewTabs.OVERVIEW,
+      isSelected: tabId === EngineViewTabs.OVERVIEW || undefined,
+      name: i18n.translate('xpack.enterpriseSearch.engines.engineView.overviewTabLabel', {
+        defaultMessage: 'Overview',
+      }),
+    },
+    {
+      append: engineData && (
+        <EuiNotificationBadge color="subdued">{engineData.indices.length}</EuiNotificationBadge>
+      ),
+      content: <div />,
+      id: EngineViewTabs.INDICES,
+      isSelected: tabId === EngineViewTabs.INDICES || undefined,
+      name: i18n.translate('xpack.enterpriseSearch.engines.engineView.indicesTabLabel', {
+        defaultMessage: 'Indices',
+      }),
+    },
+    {
+      content: <div />,
+      id: EngineViewTabs.DOCUMENTS,
+      isSelected: tabId === EngineViewTabs.DOCUMENTS || undefined,
+      name: i18n.translate('xpack.enterpriseSearch.engines.engineView.documentsTabLabel', {
+        defaultMessage: 'Documents',
+      }),
+    },
+    {
+      content: <div />,
+      id: EngineViewTabs.SCHEMA,
+      isSelected: tabId === EngineViewTabs.SCHEMA || undefined,
+      name: i18n.translate('xpack.enterpriseSearch.engines.engineView.schemaTabLabel', {
+        defaultMessage: 'Schema',
+      }),
+    },
+    {
+      content: <div />,
+      id: EngineViewTabs.PREVIEW,
+      isSelected: tabId === EngineViewTabs.PREVIEW || undefined,
+      name: i18n.translate('xpack.enterpriseSearch.engines.engineView.previewTabLabel', {
+        defaultMessage: 'Preview',
+      }),
+    },
+    {
+      content: <div />,
+      id: EngineViewTabs.API,
+      isSelected: tabId === EngineViewTabs.API || undefined,
+      name: i18n.translate('xpack.enterpriseSearch.engines.engineView.apiTabLabel', {
+        defaultMessage: 'API',
+      }),
+    },
+  ];
+  const onTabClick = (tab: EuiTabbedContentTab) => {
+    KibanaLogic.values.navigateToUrl(
+      generateEncodedPath(ENGINE_TAB_PATH, {
+        engineName,
+        tabId: tab.id,
+      })
+    );
+  };
+
+  return (
+    <EnterpriseSearchEnginesPageTemplate
+      pageChrome={[engineName]}
+      pageViewTelemetry={tabId}
+      isLoading={isLoadingEngine}
+      pageHeader={{
+        pageTitle: engineName,
+        rightSideItems: [
+          <EuiButtonEmpty
+            href={docLinks.appSearchElasticsearchIndexedEngines} // TODO: replace with real docLinks when it's created
```

Comment:
Is this TODO recorded in another issue anywhere?

Reply:
Not directly, but so far we've handled these toward the end of the release cycle as docs start to show up.